lib/ecto/migration.ex

```diff
@@ -825,6 +827,10 @@ defmodule Ecto.Migration do
         end
       end
 
+  Alternately, you can put `@disable_migration_lock true` in your migration,
+  but would mean that in a multi-node setup, multiple nodes could be running
+  the migration at once, so it is generally discommended.
```

Comment:
> Could multiple run the same migration at once, or different migrations at once?

It could be either. Before a migration is run a query is used on the migrations table to pull the completed migration versions. This is how it knows whether to ignore or proceed with a migration.

If there's no lock then you have a race condition where one node could be in the middle of completing migration 1 when another node checks the versions and see that migration 1 is not completed.

Reply:
Basically if your database doesn't support advisory locks then concurrently should be avoided or its migration should be run in an isolated environment.

Reply:
@greg-rychlewski would you say the recommendation in the post linked above is misguided?

Reply:
Also my blocking index creation does not block reads. I'm not too sure which database they are using that block reads but that is not always true.

Reply:
@greg-rychlewski thanks. Update the PR.